sui/src/rpc_gateway_client.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::rest_gateway::responses::ObjectResponse;
+use crate::rpc_gateway::{
+    Base64EncodedBytes, RpcGatewayClient as RpcGateway, SignedTransaction, TransactionBytes,
+};
+use anyhow::Error;
+use async_trait::async_trait;
+use ed25519_dalek::ed25519::signature::Signature;
```

Comment:
Wondering why this dep is required, let's assume we're not making any assumption re sig scheme (per our agility path) and use api from our crypto.rs when possible.